core/arch/arm/include/sm/sm.h

```diff
@@ -32,6 +32,10 @@ struct sm_mode_regs {
 	uint32_t und_spsr;
 	uint32_t und_sp;
 	uint32_t und_lr;
+#ifdef CFG_SM_NO_CYCLE_COUNTING
```

Comment:
It seems very convenient to store the register here, but the name of the struct has to be changed since it's not keeping registers specific for a particular mode any longer. How about `struct sm_unbanked_regs` instead?

Reply:
Sounds good, I will add a patch. Probably a good idea to rename the struct first though -- I will swap the commits when adding the review tags.